src/sentry/shared_integrations/client/proxy.py

```diff
@@ -27,26 +29,37 @@ class IntegrationProxyClient(ApiClient):  # type: ignore
 
     should_proxy_to_control = False
 
-    def __init__(self, org_integration_id: int) -> None:
+    def __init__(
+        self, integration_id: int | None = None, org_integration_id: int | None = None
+    ) -> None:
         super().__init__()
+
+        self.integration_id = integration_id
         self.org_integration_id = org_integration_id
 
         is_region_silo = SiloMode.get_current_mode() == SiloMode.REGION
-        subnet_secret = getattr(settings, "SENTRY_SUBNET_SECRET")
-        control_address = getattr(settings, "SENTRY_CONTROL_ADDRESS")
+        subnet_secret = getattr(settings, "SENTRY_SUBNET_SECRET", None)
+        control_address = getattr(settings, "SENTRY_CONTROL_ADDRESS", None)
 
         if is_region_silo and subnet_secret and control_address:
             self.should_proxy_to_control = True
             self.proxy_url = f"{settings.SENTRY_CONTROL_ADDRESS}{PROXY_BASE_PATH}"
 
     @control_silo_function
-    def authorize_request(self, request: Request) -> Request:
-        raise NotImplementedError(
-            "'authorize_request' method must be implemented to safely proxy requests."
-        )
+    def authorize_request(self, prepared_request: PreparedRequest) -> PreparedRequest:
+        """
+        Used in the Control Silo to authorize all outgoing requests to the service provider.
+        """
+        return prepared_request
 
     def finalize_request(self, prepared_request: PreparedRequest) -> PreparedRequest:
+        """
+        Every request through this subclassed clients run this method.
+        If running as a monolith/control, we must authorize each request before sending.
+        If running as a region, we don't authorize and instead, send it to our proxy endpoint.
+        """
```

Comment:
```suggestion
        If running as a region, we don't authorize and instead, send it to our proxy endpoint,
        where tokens are added in by Control Silo. We do this to avoid race conditions around
        stale tokens and centralize token refresh flows. 
        """
```